pinot-query-runtime/src/test/resources/queries/TableExpressions.json

```diff
@@ -0,0 +1,93 @@
+      {
+        "ignored": true,
+        "comments": "Relation Decorrelator not supported",
+        "sql": "SELECT * FROM {tbl} AS a WHERE a.strCol IN (SELECT b.strCol FROM {tbl} AS b WHERE b.intCol = a.intCol + 1)"
+      },
+      {
+        "ignored": true,
+        "comments": "BETWEEN with a non-deterministic single-value result from sub-query is not supported",
+        "sql": "SELECT * FROM {tbl} AS a WHERE a.intCol BETWEEN (SELECT b.intCol FROM {tbl} AS b WHERE b.intCol = a.intCol + 1) AND 100"
+      },
+      {
+        "ignored": true,
+        "comments": "Relation Decorrelator not supported",
+        "sql": "SELECT * FROM {tbl} AS a WHERE a.intCol BETWEEN (SELECT MIN(b.intCol) FROM {tbl} AS b WHERE b.intCol = a.intCol + 1) AND 100"
+      },
+      {
+        "ignored": true,
+        "comments": "EXISTS not supported",
+        "sql": "SELECT * FROM {tbl} AS a WHERE EXISTS (SELECT strCol FROM {tbl} AS b WHERE b.intCol = a.intCol + 1)"
+      },
+      {
+        "ignored": true,
+        "comments": "Relation Decorrelator not supported",
+        "sql": "SELECT * FROM {tbl} AS a WHERE (SELECT count(*) FROM {tbl} AS b WHERE b.intCol = a.intCol + 1) > 0"
+      }
+    ]
+  },
+  "group_by_and_having_tests": {
+    "psql": "7.2.3",
+    "tables": {
+      "tbl1": {
+        "schema": [
+          {"name": "strCol", "type": "STRING"},
+          {"name": "intCol", "type": "INT"}
+        ],
+        "inputs": [
+          ["a", 3],
+          ["b", 2],
+          ["c", 5],
+          ["a", 1]
+        ]
+      },
+      "tbl2": {
+        "schema": [
+          {"name": "strCol1", "type": "STRING"},
+          {"name": "strCol2", "type": "STRING"},
+          {"name": "intCol", "type": "INT"}
+        ],
+        "inputs": [
+          ["a", "foo", 1],
+          ["a", "bar", 2],
+          ["b", "alice", 42],
+          ["b", "bob", 196883]
+        ]
+      }
+    },
+    "queries": [
+      { "sql": "SELECT strCol FROM {tbl1} GROUP BY strCol" },
+      { "sql": "SELECT strCol FROM {tbl1} GROUP BY strCol, intCol" },
+      { "sql": "SELECT strCol, intCol FROM {tbl1} GROUP BY strCol, intCol" },
+      { "sql": "SELECT strCol, SUM(intCol) FROM {tbl1} GROUP BY strCol" },
+      { "sql": "SELECT strCol, b.strCol2, (sum(a.intCol) * b.intCol) AS colAlias FROM {tbl1} a INNER JOIN {tbl2} b ON a.strCol = b.strCol1 GROUP BY strCol, b.strCol2, b.intCol" },
+      { "sql": "SELECT MOD(b.intCol, 2), sum(a.intCol) AS colAlias FROM {tbl1} a INNER JOIN {tbl2} b ON a.strCol = b.strCol1 GROUP BY MOD(b.intCol, 2)" },
+      { "sql": "SELECT strCol, SUM(intCol) FROM {tbl1} GROUP BY strCol HAVING SUM(intCol) > 3" },
+      { "sql": "SELECT strCol, SUM(intCol) FROM {tbl1} GROUP BY strCol HAVING AVG(intCol) > 1 AND MIN(intCol) < 10" },
+      { "sql": "SELECT strCol, b.strCol2, (sum(a.intCol) * b.intCol) AS colAlias FROM {tbl1} a INNER JOIN {tbl2} b ON a.strCol = b.strCol1 GROUP BY strCol, b.strCol2, b.intCol HAVING avg(a.intCol) < 100" }
```

Comment:
Let's also add tests for subquery in the SELECT clause as a scalar operand ?

Something like ....

```
SELECT 
empID, lastName, 
firstName, (
       CASE WHEN deptID = 
                 (
                     SELECT deptID 
                     FROM departments 
                     WHERE ......
                 ) 
         THEN 'blah1' 
         ELSE 'blah2'
         END)
FROM emp
```